mlflow/langchain/databricks_dependencies.py

```diff
@@ -121,6 +174,7 @@ def _extract_dependency_list_from_lc_model(lc_model) -> Generator[Resource, None
     yield from _extract_databricks_dependencies_from_chat_model(lc_model)
     yield from _extract_databricks_dependencies_from_retriever(lc_model)
     yield from _extract_databricks_dependencies_from_llm(lc_model)
+    yield from _extract_databricks_dependencies_from_agent_tools(lc_model)
```

Comment:
If its an agent, does it need to be part of the 
```
for attr_name in _LEGACY_MODEL_ATTR_SET:
        yield from _extract_dependency_list_from_lc_model(getattr(lc_model, attr_name, None))
```

This recursive call?
Can it run as its own function if its an agent?